engine-tests/src/tests/repro.rs

```diff
@@ -27,7 +27,7 @@ fn repro_GdASJ3KESs() {
         block_timestamp: 1_645_717_564_644_206_730,
         input_path: "src/tests/res/input_GdASJ3KESs.hex",
         evm_gas_used: 706_713,
-        near_gas_used: 122,
+        near_gas_used: 124,
```

Comment:
It's interesting that this increases all the transactions here consistently by 2 Tgas. I think this could be a big enough issue that we want to address it. I do not think it is acceptable to have all transactions cost 2 Tgas more in production. I think there two ways we can fix this issue:

1. Do a state migration. If we read the old state format then we re-write the state to be in the new format. This will increase the gas cost even more for the first transaction, but then it should go back to normal after that. The way to test the impact of this approach would be to include a dummy transaction in `repro_common` before the main transaction is executed. The state migration would happen in that first transaction, then we would see what the new "typical" case is in the benchmarks.
2. Write the backwards-compatible deserialization logic in a more clever way. I assume the 2 Tgas cost comes from having to deserialize the state twice (first attempt the new enum, it fails so then attempt the legacy version). But we don't have to do it that way. Another approach would be to take advantage of what we know about `EngineState` values in the wild. In particular, we know that the first field is the 32-byte chain id and that chain-ids in practice are 64-bit (8-byte) numbers; therefore the first byte of the `EngineState` serialization will always be 0. So we could reserve 0 in the new enum format (i.e. don't use it as a variant discriminator byte), and then the check for the legacy format is trivial: if the first byte is 0 then it is in the legacy format, if it is non-zero then it is the new enum format.

The second approach is clever, but a little hack in my opinion, so I think it would be better to try the first approach.